File: frontend/fixme_app/lib/models/conversation.dart
class Conversation {
  final int id;
  final int serviceRequestId;
  final int customerId;
  final int providerId;

  Conversation({
    required this.id,
    required this.serviceRequestId,
    required this.customerId,
    required this.providerId,
  });

  factory Conversation.fromJson(Map<String, dynamic> j) => Conversation(
        id: (j['id'] as num).toInt(),
        serviceRequestId: (j['serviceRequestId'] as num).toInt(),
        customerId: (j['customerId'] as num).toInt(),
        providerId: (j['providerId'] as num).toInt(),
      );
}

File: backend/auth-service/src/main/java/com/fixme/authservice/model/Conversation.java
package com.fixme.authservice.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(
        name = "conversations",
        uniqueConstraints = @UniqueConstraint(columnNames = "service_request_id")
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Conversation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false)
    @JoinColumn(name = "service_request_id")
    private ServiceRequest serviceRequest;

    @ManyToOne(optional = false)
    @JoinColumn(name = "customer_id")
    private User customer;

    @ManyToOne(optional = false)
    @JoinColumn(name = "provider_id")
    private User provider;

    private LocalDateTime createdAt;
}
